lib/genome/groupers/drug_grouper.rb

```diff
@@ -1,154 +1,149 @@
-              drug_claim.drug = drug if drug_claim.drug.nil?
-              drug_claim.save
-            end
-          else
-            DataModel::Drug.new.tap do |g|
-              g.name = key
-              g.drug_claims = drug_claims
-              g.save
-            end
-          end
+        # check drug aliases (indirect)
+        #   - against claim name
+        #   - against claim aliases
+
+        if (drug_id = DataModel::DrugAlias.where('upper(alias) in (?)', drug_claim.names).pluck(:drug_id).to_set).one?
+          drug = DataModel::Drug.where(id: drug_id.first).first
+          add_drug_claim_to_drug(drug_claim, drug)
+          return drug_claim
+        elsif drug_id.many?
+          indirect_multimatch << drug_claim
+          return nil
         end
+
       end
 
-      def self.add_members
-        DataModel::DrugClaim.all.each do |drug_claim|
-          next unless drug_claim.drug.nil?
-          indirect_groups = Hash.new { |h, k| h[k] = 0 }
-          direct_groups = Hash.new { |h, k| h[k] = 0 }
-          direct_groups[drug_claim.name] += 1 if DataModel::Drug.where(name: drug_claim.name).any?
-          drug_claim.drug_claim_aliases.each do |drug_claim_alias|
-            direct_groups[drug_claim_alias.alias] +=1 if DataModel::Drug.where(name: drug_claim_alias.alias).any?
-            alt_drugs = @alt_to_other[drug_claim_alias.alias].map(&:drug_claim)
-            alt_drugs.each do |alt_drug|
-              indirect_drug = alt_drug.drug
-              indirect_groups[indirect_drug.name] += 1 if indirect_drug
-            end
-            nomenclature = drug_claim_alias.nomenclature
-            if nomenclature =~ /pubchem.*(compound)|(cid)/i
-              alt_drugs = @alt_to_pubchem_cid[drug_claim_alias.alias].map(&:drug_claim)
-              alt_drugs.each do |alt_drug|
-                indirect_drug = alt_drug.drug
-                indirect_groups[indirect_drug.name] += 1 if indirect_drug
-              end
-            end
-          end
+      def direct_multimatch
+        @direct_multimatch ||= Set.new
+      end
 
-          if direct_groups.keys.length == 1
-            drug = DataModel::Drug.where(name: direct_groups.keys.first).first
-            drug.drug_claims << drug_claim unless drug.drug_claims.include?(drug_claim)
-            drug.save
-          elsif direct_groups.keys.length == 0 && indirect_groups.keys.length == 1
-            drug = DataModel::Drug.where(name: indirect_groups.keys.first).first
-            drug.drug_claims << drug_claim unless drug.drug_claims.include?(drug_claim)
-            drug.save
-          end
+      def molecule_multimatch
+        @molecule_multimatch ||= Set.new
+      end
+
+      def indirect_multimatch
+        @indirect_multimatch ||= Set.new
+      end
+
+      def drug_claims_not_in_groups
+        DataModel::DrugClaim.where(drug_id: nil).to_a.keep_if do |drug_claim|
+          !(
+            direct_multimatch.member? drug_claim or
```

Comment:
Generally speaking, you'll want to use `||` rather than `or` here. They are similar, but not actually equivalent in ruby. The english words bind at a lower precedence and are more appropriately used for say, flow control `do_something or raise Exception` and the symbolic ones are more appropriate for boolean logic like this.